File: src/test/java/org/micoli/php/builders/PostToggleActionBuilder.java
package org.micoli.php.builders;

import org.micoli.php.tasks.configuration.runnableTask.postToggle.PostToggleShell;

public class PostToggleActionBuilder {
    private final PostToggleShell action;

    private PostToggleActionBuilder() {
        this.action = new PostToggleShell();
    }

    public static PostToggleActionBuilder create() {
        return new PostToggleActionBuilder();
    }

    public PostToggleActionBuilder withId(String id) {
        action.id = id;
        return this;
    }

    public PostToggleActionBuilder withLabel(String label) {
        action.label = label;
        return this;
    }

    public PostToggleActionBuilder withCommand(String command) {
        action.command = command;
        return this;
    }

    public PostToggleActionBuilder withCwd(String cwd) {
        action.cwd = cwd;
        return this;
    }

    public PostToggleShell build() {
        return action;
    }
}

File: src/test/kotlin/org/micoli/php/builders/PostToggleActionBuilder.kt
package org.micoli.php.builders

import org.micoli.php.tasks.configuration.runnableTask.postToggle.PostToggleShell

class PostToggleActionBuilder private constructor() {
    private val action: PostToggleShell = PostToggleShell()

    fun withId(id: String): PostToggleActionBuilder {
        action.id = id
        return this
    }

    fun withLabel(label: String): PostToggleActionBuilder {
        action.label = label
        return this
    }

    fun withCommand(command: String): PostToggleActionBuilder {
        action.command = command
        return this
    }

    fun withCwd(cwd: String): PostToggleActionBuilder {
        action.cwd = cwd
        return this
    }

    fun build(): PostToggleShell = action

    companion object {
        fun create(): PostToggleActionBuilder = PostToggleActionBuilder()
    }
}
